configuration/linux-capability.md

```diff
@@ -0,0 +1,182 @@
+
+    $ fluent-gem install capng_c
+
+Or install it yourself as for td-agent:
+
+    $ td-agent-gem install capng_c
+
+**Note:** capng_c uses `pkg-config` to link libcap-ng library. If you couldn't handle Linux capability with capng_c installation, please confirm `pgk-config` package is installed on your box.
+
+## Capability handling on in_tail
+
+Currently, `in_tail` which is the one of the Fluentd core plugin handles the following Linux capabilities:
+
+* `CAP_DAC_READ_SEARCH` (`:dac_read_search` on `in_tail` code.)
+* `CAP_DAC_OVERRIDE` (`:dac_override` on `in_tail` code.)
+
+Set up `cap_dac_read_search` or `cap_dac_override` to using Ruby executable:
+
+### Using CAP_DAC_READ_SEARCH
+
+```console
+$ sudo setcap cap_dac_read_search=+eip /path/to/bin/ruby
+```
+
+### Using CAP_DAC_OVERRIDE
+
+```console
+$ sudo setcap cap_dac_override=+eip /path/to/bin/ruby
+```
+
+**Note:** Under rbenv environment, `which ruby` returns shell script wrapper. If users want to set capability on rbenv-ed Ruby, please use the following command:
+
+```console
+$ sudo setcap YOUR_USING_CAPABILITY=+eip $(rbenv prefix)/bin/ruby
+```
+
+#### Example setting up capability for rbenv-ed Ruby
+
+```console
+$ sudo setcap cap_dac_override,cap_dac_read_search=+eip $(rbenv prefix)/bin/ruby
+$ filecap $(rbenv prefix)/bin/ruby
+file                 capabilities
+/home/fluentd/.rbenv/versions/2.6.3/bin/ruby     dac_override, dac_read_search
+```
+
+### Actual Example for Linux capability handling in in_tail
+
+When adding `cap_dac_override` (partial privileges for rw file) and `cap_dac_read_search` (partial privileges for read only), Fluentd/td-agent can handle to read 640 permission files such as `/var/log/syslog`:
+
+```console
+$ ls -lh /var/log/syslog
+-rw-r----- 1 syslog adm 29K Nov  5 14:35 /var/log/syslog
+```
+
+This file cannot read form ordinal users:
+
+```console
+$ cat /var/log/syslog
+cat: /var/log/syslog: Permission denied
+```
+
+Attach `cap_dac_read_search` for using Ruby executable binary:
+
+```console
+$ sudo setcap cap_dac_read_search=+eip /path/to/bin/ruby
+$ filecap /path/to/bin/ruby
+file                 capabilities
+/path/to/bin/ruby     dac_read_search
+```
+
+And prepare the following configuration:
+
+```aconf
+<source>
+  @type tail
+  path /var/log/syslog
+  pos_file /var/run/fluentd/log/syslog_test_with_capability.pos
+  tag test
+  rotate_wait 5
+  read_from_head true
+  refresh_interval 60
+  <parse>
+    @type syslog
+  </parse>
+</source>
+
+<match test>
+  @type stdout
+</match>
+```
+
+Make and change ownership directory:
+
+```console
+$ sudo mkdir /var/run/fluentd
+$ sudo chown `whoami` /var/run/fluentd
+```
+
+Then, run as ordinal user with `cap_dac_read_search` capability attached Ruby:
+
+```console
+$ bundle exec fluentd -c in_tail_camouflage_permission.conf
+2020-11-05 14:47:57 +0900 [info]: parsing config file is succeeded path="example/in_tail.conf"
+2020-11-05 14:47:57 +0900 [info]: gem 'fluentd' version '1.11.4'
```

Comment:
version should be `1.12.0`.

Reply:
Oh! :fearful:

Reply:
Fixed https://github.com/fluent/fluentd-docs-gitbook/pull/248/commits/3441bbe75cc492ab77eb87e0143b8d65ac099f7e